### PR TITLE
CI/CD refactor

### PR DESCRIPTION
**Pull request by itaysk**
This PR has a few goals:
- fix https://github.com/aquasecurity/trivy/issues/205
- upgrade goreleaser version and allow us to keep up to date in the future
- push to docker hub via goreleaser instead of docker hub build
- unify build processes (one is in dockerfile ine is in goreleaser) for consistent artifacts

I tested everything on my fork using replicas docker-hub/circle-ci/brew-tap

Note the following tasks that should be done in concert with merging this PR:

- create a repo in `aquasec` docker hub for `trivy-ci`
- disable the dockerhub build for `aquasec/trivy`
- add `$DOCKER_USER` and `$DOCKER_PASS` secrets to circle ci 

@knqyf263 @eranbibi

### CODE REVIEW

File: ci/Dockerfile [hunk 1]
```diff
@@ -1,20 +1,10 @@
-FROM bepsays/ci-goreleaser:1.12-2
+FROM circleci/golang:1.12-buster
```
Comment (knqyf263) [inline ci/Dockerfile]: Could you use 1.13-buster?

File: ci/Dockerfile [hunk 2]
```diff
@@ -1,20 +1,10 @@
...
-    # Darwin
-    && make clean \
-    && ../dist/configure --prefix=/usr/local --host=x86_64-apple-darwin15 \
-    && make \
-    && make install
+ARG GORELEASER_VERSION=0.110.0
+ARG GORELEASER_ARTIFACT=goreleaser_Linux_x86_64.tar.gz
+RUN wget https://github.com/goreleaser/goreleaser/releases/download/v${GORELEASER_VERSION}/${GORELEASER_ARTIFACT} \
+    && sudo tar -xzf ${GORELEASER_ARTIFACT} -C /usr/bin/ goreleaser \
+	&& rm ${GORELEASER_ARTIFACT}
```
Comment (knqyf263) [inline ci/Dockerfile]:
```suggestion
    && rm ${GORELEASER_ARTIFACT}
```

File: goreleaser.yml [hunk 3]
```diff
@@ -22,60 +23,78 @@ builds:
...
+    dependencies:
+      - rpm
+    homepage: "https://github.com/aquasecurity/trivy"
+    description: ""
+    test: |
+      system "#{bin}/program --version"
+
+dockers:
+  - image_templates:
+      - "docker.io/aquasec/trivy:{{ .Version }}"
```
Comment (knqyf263) [inline goreleaser.yml]: Can you add `latest` tag as well?
```
- "docker.io/aquasec/trivy:latest"
```

File: ci/Dockerfile (hunk 1 above)
Comment (itaysk) [reply ci/Dockerfile]: I think upgrading go version is another PR and needs more testing

File: goreleaser.yml (hunk 3 above)
Comment (itaysk) [reply goreleaser.yml]: added

File: ci/Dockerfile (hunk 1 above)
Comment (knqyf263) [reply ci/Dockerfile]: I've already worked on supporting Go 1.13. I think it should work with Go 1.13. But another PR is OK.

**Pull request merged**